assets/js/payment-methods-demo/payment-methods/stripe/payment-method.js

```diff
@@ -0,0 +1,317 @@
+		setSource( null );
+
+		// @todo we'll want to do inline invalidation errors for any element
+		// inputs
+	};
+	const updateCardBrand = ( brand ) => {
+		// @todo this updates the card brand logo with non-inline CC forms
+		// (see function in stripe.js)
+	};
+	const onChange = ( paymentEvent ) => {
+		if ( paymentEvent.elementType === 'cardNumber' ) {
+			updateCardBrand( paymentEvent.brand );
+		}
+		if ( paymentEvent.error ) {
+			onStripeError( paymentEvent );
+		} else {
+			setSource( null );
+		}
+	};
+	const renderedCardElement = getStripeServerData().inline_cc_form ? (
+		<CardElement
+			style={ elementStyles }
+			onChange={ onChange }
+			hidePostalCode={ true }
+		/>
+	) : (
+		<CardElements
+			style={ elementStyles }
+			classes={ elementClasses }
+			onChange={ onChange }
+		/>
+	);
+	useEffect( () => {}, [] );
+
+	const createSource = async ( stripeBilling ) => {
+		return await stripe.createSource( stripeBilling );
+	};
+	const onSubmit = () => {
+		const { billingData } = billing;
+		const stripeBilling = {
+			address: {
+				line1: billingData.address_1,
+				line2: billingData.address_2,
+				city: billingData.city,
+				state: billingData.state,
+				postal_code: billingData.postcode,
+				country: billingData.country,
+			},
+		};
+		if ( billingData.phone ) {
+			stripeBilling.phone = billingData.phone;
+		}
+		if ( billingData.email ) {
+			stripeBilling.email = billingData.email;
+		}
+		if ( billingData.first_name || billingData.last_name ) {
+			stripeBilling.name = `${ billingData.first_name } ${ billingData.last_name }`;
+		}
+
+		const response = createSource( stripeBilling );
+		if ( response.error ) {
+			// @todo set payment error
+			// return error object including any validation errors if present?
+		}
+		paymentStatus.setPaymentStatus().success( billingData, {
+			sourceId: response.source.id,
+		} );
+
+		// this should hook into checkout processing? Basically when the checkout
+		// processing button is clicked
+		// if saved source (card) is selected by user, then there should be no
+		// processing/validation by stripe.
+		// as a part of the submission, go to `createSource`
+
+		// if submission completes okay, then set payment status to complete
+		// and that's where billing data and payment data are passed through.
+	};
+	const onComplete = () => {
+		// @todo this is invoked on checkout complete success
+		// in this place just set the currentPaymentStatus to completed.
+	};
+	const onError = () => {
+		// hooked into checkout complete error event. Not sure if will be needed.
+	};
+	// hook into and register callbacks for events.
+	useEffect( () => {
+		// validation and stripe processing (get source etc).
+		const unsubscribeProcessing = eventRegistration.onCheckoutProcessing(
+			onSubmit
+		);
+		const unsubscribeCheckoutComplete = eventRegistration.onCheckoutCompleteSuccess(
+			onComplete
+		);
+		const unsubscribeCheckoutCompleteError = eventRegistration.onCheckoutCompleteError(
```

Comment:
I'm considering changing the name of this event in the checkout context to `onCheckoutProcessingError`, and then `onCheckoutCompleteSuccess/Error` would just need to be `onCheckotuComplete`.